be/src/util/buffered_stream.cpp

```diff
@@ -64,4 +66,87 @@ Status BufferedInputStream::_read_data() {
     return Status::OK();
 }
 
+Status DefaultBufferedInputStream::get_bytes(const uint8_t** buffer, size_t offset, size_t* nbytes) {
+    seek_to(offset);
+    return get_bytes(buffer, nbytes, false);
+}
+
+// ===================================================================================
+
+SharedBufferedInputStream::SharedBufferedInputStream(RandomAccessFile* file) : _file(file) {}
+
+Status SharedBufferedInputStream::set_io_ranges(const std::vector<IORange>& ranges) {
+    if (ranges.size() == 0) {
+        return Status::OK();
+    }
+    std::vector<IORange> check(ranges);
+    std::sort(check.begin(), check.end());
+    // check io range is not overlapped.
+    for (size_t i = 1; i < ranges.size(); i++) {
+        if (check[i].offset < (check[i - 1].offset + check[i - 1].size)) {
```

Comment:
Why not merge overlapping ranges here to reduce duplicate data requests？

Reply:
Two reasons:
1. I have thought clearly about the cases if there are overalpped ranges, that if there are some buggy cases. I'm not sure.
2. From application view, there will be no overlapped range. Because in a row group, column data are not overalapped.

Reply:
OK